### ISSUE TITLE
Enable Commit to cover new proposal types

### ISSUE THREAD

**Issue by bifurcation** [mlswg/mls-protocol#383]
In principle, the set of ProposalType values is extensible.  But with the current Commit syntax, there is no way for a Commit to reference a Proposal with a type other than Add/Update/Remove.  We could enable this by adding a field to the commit that would carry such references:

```
struct {
    ProposalID updates<0..2^32-1>;
    ProposalID removes<0..2^32-1>;
    ProposalID adds<0..2^32-1>;
    ProposalID others<0..2^32-1>;    // <------

    optional<DirectPath> path;
} Commit;
```

However, if other proposal types had sequencing requirements, they would not get the benefit of explicit, separate grouping and ordering as is done for the current proposal types.

If we don't extend the Commit in this way, we should add some explanatory text to clarify that new proposal types require an extension that changes the format of the Commit message.

**Comment by bifurcation** [comment]
Personally, I lean toward the "do nothing" option here.  If we take the view that proposals should describe modifications to the tree (other actions being signaled in [Commit extensions](https://github.com/mlswg/mls-protocol/issues/366)), then Add, Update, and Remove are pretty much the available atomic operations.  You could envision some minor variations (an Add with a specified location, say) or compositions (Resync = Remove+Add), but nothing is coming to my mind that would be important enough to merit the extra complexity.

**Comment by bifurcation** [comment]
Fixed by #389
